### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by AHP15**
In this PR, I added authentication using devise

- Can register a new user.
- User logs in with a combination of email and password.
- Hashed password should be stored in the database.
- Ask for confirmation of email.
- Can reset password.

### CODE REVIEW

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @Abdessittir , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request :red_circle: 

You have submitted a project, that does not have a `commit` from your pair programming partner. Please note that on this milestone you are supposed to work with your pair programming partner and each of you needs to commit to both repositories. Please reach out to your partner and let him/her contribute to some part of your project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password %>
+    <% if @minimum_password_length %>
+    <em>(<%= @minimum_password_length %> characters minimum)</em>
+    <% end %><br />
+    <%= f.password_field :password, autocomplete: "new-password" %>
+  </div>
```
Comment (TadesseDev) [inline app/views/devise/registrations/new.html.erb:17]: - [ ] Hello Team, Please notice though your app is registering a user there is an error `NameError in Devise::RegistrationsController#create` which rises immediately as a user sign up. Please address this for a better user experience. 

https://user-images.githubusercontent.com/69077061/223696268-8e0aafa1-79af-4b2d-bfa1-f723ec5d1de6.mp4

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+<%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password %><br />
+    <%= f.password_field :password, autocomplete: "current-password" %>
+  </div>
```
Comment (TadesseDev) [inline app/views/devise/sessions/new.html.erb:12]: - [ ] For your login, Please make sure a user can log in with a combination of email and password. Your application asks to confirm my email before signing in (which is good), but It won't send a confirmation email.

https://user-images.githubusercontent.com/69077061/223698451-82ba7e8a-77dd-40ee-875a-ca355c4738f1.mp4

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [ ]  Correct GitFlow  ✅
 - [ ]  Good commit history  ✅
 - [ ]  Good PR title/description  ✅
 - [ ]  All linters passed  ✅
 
## Required Changes ♻️
- [ ] You have done a great job Team. But for this project, we need to make changes to your `application_controller` and remove the `current_user` action, which you have not done. I'm incapable of giving you an inline comment because you haven't made a change to your controller, hence not reflected on the `file_change` tab. But I have added the permanent link for the file to be changed. 
https://github.com/Abdessittir/blog-app/blob/e7bbb11b9a4b34b58022fbc47723ef5d14c9779f/app/controllers/application_controller.rb#L1-L5
- [ ] Also, Good job adding the MIT file, The problem is you need to update its content in regard to your name and project title. Please replace the `[YOUR NAME]` part with your name, `[APP TYPE]` with your app type, and the [YEAR] with the current year. 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**